CalibCalorimetry/HcalPlugins/src/HcalHardcodeCalibrations.cc

```diff
@@ -450,37 +451,41 @@ std::unique_ptr<HcalRespCorrs> HcalHardcodeCalibrations::produceRespCorrs (const
  
   auto result = std::make_unique<HcalRespCorrs>(topo);
   std::vector <HcalGenericDetId> cells = allCells(*topo, dbHardcode.killHE());
-  for (std::vector <HcalGenericDetId>::const_iterator cell = cells.begin (); cell != cells.end (); ++cell) {
+  for (const auto& cell : cells) {
 
     double corr = 1.0; 
 
-    if ((he_recalibration != 0 ) && 
-	((*cell).genericSubdet() == HcalGenericDetId::HcalGenEndcap)) {
-      
-      int depth_ = HcalDetId(*cell).depth();
-      int ieta_  = HcalDetId(*cell).ieta();
-      corr = he_recalibration->getCorr(ieta_, depth_); 
-      
-      /*
-	std::cout << "HE ieta, depth = " << ieta_  << ",  " << depth_  
-	<< "   corr = "  << corr << std::endl;
-      */
-
-    }
-    else if ((hf_recalibration != 0 ) && 
-	((*cell).genericSubdet() == HcalGenericDetId::HcalGenForward)) {   
-      int depth_ = HcalDetId(*cell).depth();
-      int ieta_  = HcalDetId(*cell).ieta();
+    //check for layer 0 reweighting: when depth 1 has only one layer, it is layer 0
+    if( useLayer0Weight && 
+      ((cell.genericSubdet() == HcalGenericDetId::HcalGenEndcap) || (cell.genericSubdet() == HcalGenericDetId::HcalGenBarrel)) &&
+      (HcalDetId(cell).depth()==1 && dbHardcode.getLayersInDepth(HcalDetId(cell).ieta(),HcalDetId(cell).depth(),topo)==1) )
```

Comment:
Yep, that's it. I actually thought about that for a second earlier, and decided that I couldn't have been so dumb as to have an absolute value issue there, but it turns out that I was. Will fix now...